src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/ManageHearingsNotificationRequestMapper.java

```diff
@@ -18,21 +19,88 @@
 @RequiredArgsConstructor
 public class ManageHearingsNotificationRequestMapper {
 
+    private record PartySpecificDetails(
+        String recipientEmailAddress,
+        String recipientName
+    ) {}
+
     /**
      * Constructs a {@link NotificationRequest} for sending a hearing notification to the applicant's solicitor.
      *
-     * <p>This method gathers relevant details from the case data and hearing, including email, names,
-     * reference numbers, case type, and selected court. It uses these values to build a
-     * {@link NotificationRequest} that is suitable for the Notify API.</p>
+     * <p>Gets party specific details for an applicant, then passes to
+     * buildHearingNotificationForParty to fill in the rest from data common to parties. </p>
      *
      * @param finremCaseDetails the case details including case ID, type, and applicant/solicitor info
-     * @param hearing the hearing information, specifically used for the hearing type
-     * @return a fully constructed {@link NotificationRequest} ready for dispatch
+     * @param hearing the hearing information
+     * @return a fully constructed {@link NotificationRequest}
      */
     public NotificationRequest buildHearingNotificationForApplicantSolicitor(
             FinremCaseDetails finremCaseDetails,
             Hearing hearing) {
 
+        PartySpecificDetails partySpecificDetails = new PartySpecificDetails(
+                finremCaseDetails.getData().getAppSolicitorEmail(),
+                nullToEmpty(finremCaseDetails.getData().getAppSolicitorName())
+        );
+
+        return buildHearingNotificationForParty(finremCaseDetails, hearing, partySpecificDetails);
+    }
+
+    /**
+     * Constructs a {@link NotificationRequest} for sending a hearing notification to the respondent's solicitor.
+     *
+     * <p>Gets party specific details, then passes to
+     * buildHearingNotificationForParty to fill in the rest from data common to parties. </p>
+     *
+     * @param finremCaseDetails the case details including case ID, type, and party solicitor info
+     * @param hearing the hearing information
+     * @return a fully constructed {@link NotificationRequest}
+     */
+    public NotificationRequest buildHearingNotificationForRespondentSolicitor(
+            FinremCaseDetails finremCaseDetails,
+            Hearing hearing) {
+
+        PartySpecificDetails partySpecificDetails = new PartySpecificDetails(
+                finremCaseDetails.getData().getRespondentSolicitorEmailForContested(),
+                nullToEmpty(finremCaseDetails.getData().getRespondentSolicitorName())
+        );
+
+        return buildHearingNotificationForParty(finremCaseDetails, hearing, partySpecificDetails);
+    }
+
+    /**
+     * Work in progress. Intervener bug means interveners never on the party list.
```

Comment:
should we add a TODO text?

Reply:
We could.  But we've already a few todos in FR, so this is unique and I can find it more easily ;-)